internal/stats/aggregate.go

```diff
@@ -0,0 +1,212 @@
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type AggregationQuery struct {
+	EnableAggregation bool     `json:"enable-aggregation,omitempty"`
+	GroupByFields     []string `json:"group-by,omitempty"`
+	AggregateField    string   `json:"aggregate,omitempty"`
+}
+
+type QueryResponse struct {
+	Response []map[string]any `json:"response"`
+}
```

Comment:
Seems like either a strange location or name for this struct.
This file should generally just be about aggregation.